Source/WTF/wtf/Assertions.h

```diff
@@ -541,6 +541,19 @@ constexpr bool assertionFailureDueToUnreachableCode = false;
 #endif
 #endif
 
+#ifdef __cplusplus
+namespace WTF {
+template <typename T>
+struct AlwaysFalse {
+    static constexpr bool value = false;
+};
+} // namespace WTF
+#define STATIC_ASSERT_NOT_REACHED_WITH_DEPENDENT_TYPE(T) do { \
```

Comment:
Should we give this macro a string parameter to pass to static_assert, so you get a nice error message?